main.go

```diff
@@ -73,6 +99,19 @@ func checkFiles(w io.Writer, rootRule rules.Rule, cfg *config.Config, files []so
 		// `allowUnmatched` is explicitly set to `true` in the configuration.
 		if !result.Success || (result.Unmatched.Size() > 0 && !cfg.AllowUnmatched) {
 			problematicFiles = append(problematicFiles, file.Path)
+
+			sarifResult := rules.FormatSarif(result)
+			sarifResult.File = file.Path
```

Comment:
We probably should pass the file path as argument to `FormatSarif` instead of producing a value that is incomplete until we add the file info separately.

Reply:
fixed